docs/contributing/writing-instrumentation.md

```diff
@@ -46,85 +46,314 @@ Within the subfolder, create three folders `library` (skip if library instrument
-spans as part of the implementation of an interceptor for the library. The module should generally
-only depend on the OpenTelemetry API, `instrumentation-common`, and the instrumented library itself.
-[instrumentation-library.gradle](../../gradle/instrumentation-library.gradle) needs to be applied to
-configure build tooling for the library.
+Let's start with the library instrumentation. Create the `build.gradle.kts` file in the `library`
+directory:
+
+```kotlin
+plugins {
+  id("otel.library-instrumentation")
+}
+```
+
+The `otel.library-instrumentation` gradle plugin will apply all the default settings and configure
+build tooling for the library instrumentation module.
+
+Now let's move on to the actual Java code - by convention, our library instrumentations are centered
+around `*Tracing` and `*TracingBuilder` classes. These two are usually the only public classes in
+the whole module; we want to keep the number of public classes/methods as small as possible.
+
+Start by creating a `YarpcTracing` class:
+
+```java
+public final class YarpcTracing {
+
+  public static YarpcTracing create(OpenTelemetry openTelemetry) {
+    return builder(openTelemetry).build();
+  }
+
+  public static YarpcTracingBuilder builder(OpenTelemetry openTelemetry) {
+    return new YarpcTracingBuilder(openTelemetry);
+  }
+
+  // ...
+
+  YarpcTracing() {}
+
+  public Interceptor newTracingInterceptor() {
+    // ...
+  }
+}
+```
+
+By convention, the `YarpcTracing` class exposes the `create()` and `builder()` methods as the only
+way of constructing a new instance; the constructor must be kept package-private (at most). Most of
+the configuration/construction logic happens in the builder class, and we don't want to expose any
+other way of creating a new instance other than using the builder.
+
+The `newTracingInterceptor()` method listed in the example code will return an implementation of one
+of the library's interfaces that'll actually add the telemetry. This part will look different for
+every instrumented library: some of them expose interceptor/listener interfaces that can be easily
+plugged into the library, some of them have a library interface that we can use to implement a
+decorator that emits telemetry when used.
+
+Let's take a closer look at the builder class next:
+
+```java
+public final class YarpcTracingBuilder {
+
+  YarpcTracingBuilder(OpenTelemetry openTelemetry) {}
+
+  // ...
+
+  public YarpcTracing build() {
+    // ...
+  }
+}
+```
+
+The builder must have a package-private constructor (so that the only way of creating a new one is
+calling the `YarpcTracing#builder()` method) and a public `build()` method that will return a new,
+properly configured `YarpcTracing` instance.
+
+Aside from that, the library instrumentation builders may contain configuration knobs that allow to
+customize the behavior of the instrumentation. Most of these knobs are used to configure the
```

Comment:
First time I read about "knobs". Can we replace it with something simpler, such as "settings" or "parameters"?

Reply:
WDYT about "options"?